hera_pspec/uvpspec.py

```diff
@@ -1497,28 +1503,35 @@ def combine_uvpspec(uvps, verbose=True):
     Nblpairts = len(new_blpts)
     Npols = len(new_pols)
 
+
+    #store covariance only if all uvps have stored covariance.
+    u.store_cov = np.all(np.array([uvp.store_cov for uvp in uvps]))
     # create new empty data arrays and fill spw arrays
     u.data_array = odict()
     u.integration_array = odict()
     u.wgt_array = odict()
     u.nsample_array = odict()
+    if u.store_cov:
+        u.cov_array = odict()
     u.scalar_array = np.empty((Nspws, Npols), np.float)
     u.freq_array, u.spw_array, u.dly_array = [], [], []
     for i, spw in enumerate(new_spws):
         u.data_array[i] = np.empty((Nblpairts, spw[2], Npols), np.complex128)
         u.integration_array[i] = np.empty((Nblpairts, Npols), np.float64)
         u.wgt_array[i] = np.empty((Nblpairts, spw[2], 2, Npols), np.float64)
         u.nsample_array[i] = np.empty((Nblpairts, Npols), np.float64)
+        if u.store_cov:
+            u.cov_array[i]=np.empty((Nblpairts,spw[2],spw[2],Npols),np.complex128)
```

Comment:
spaces between operator and after commas:
```
u.cov_array[i] = np.empty((Nblpairts, spw[2], spw[2], Npols), dtype=np.complex128)
```

Reply:
commas added.